### PR TITLE
6.3 hardening 6

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java [hunk 1]
```diff
@@ -428,7 +428,9 @@
 
   /**
    * @since 4.0
+   * @deprecated no more used since 6.3. See https://jira.sonarsource.com/browse/SONAR-8610
    */
+  @Deprecated
   String HOURS_IN_DAY = "sonar.technicalDebt.hoursInDay";
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/utils/Durations.java [hunk 2]
```diff
@@ -62,76 +60,84 @@ public Duration create(long minutes) {
...
   /**
    * Return the formatted work duration.
    * <br>
-   * Example : format(Locale.FRENCH, Duration.encode("9d 10h"), DurationFormat.SHORT) -&gt; 10j 2h (if sonar.technicalDebt.hoursInDay property is set to 8)
+   * Example : format(Locale.FRENCH, Duration.encode("9d 10h"), DurationFormat.SHORT) -&gt; 10d 2d
    *
+   * @deprecated since 6.3 as the {@link Locale#ENGLISH} is always used. Use {@link #format(Duration)} instead
    */
+  @Deprecated
   public String format(Locale locale, Duration duration) {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/utils/Durations.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/utils/Durations.java [hunk 3]
```diff
@@ -62,76 +60,84 @@ public Duration create(long minutes) {
...
 
   /**
    * Return the formatted work duration.
-   * <br>
-   * Example : format(Locale.FRENCH, Duration.encode("9d 10h"), DurationFormat.SHORT) -&gt; 10j 2h (if sonar.technicalDebt.hoursInDay property is set to 8)
    *
+   * @deprecated since 6.3 as the {@link Locale#ENGLISH} is always used. Use {@link #format(Duration)} instead
    */
+  @Deprecated
   public String format(Locale locale, Duration duration, DurationFormat format) {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/utils/Durations.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/utils/Durations.java [hunk 4]
```diff
@@ -36,6 +32,16 @@
...
+  private static final String HOURS_FORMAT = "%sh";
+  private static final String DAYS_FORMAT = "%sd";
+
+  private static final int HOURS_IN_DAY = 8;
+
+  /**
+   * @deprecated since 6.3, only one format is available
+   */
+  @Deprecated
   public enum DurationFormat {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/utils/Durations.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 1 major
* ![INFO][INFO] 4 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WebServiceEngine.java#L133](https://github.com/SonarSource/sonarqube/blob/83259cd40c6bd08d7483b6b22a8a7a4dd9e9217f/server/sonar-server/src/main/java/org/sonar/server/ws/WebServiceEngine.java#L133): Make "sendErrors" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: server/sonar-server/src/main/java/org/sonar/server/ws/WebServiceFilter.java [hunk 5]
```diff
@@ -21,42 +21,49 @@
...
-      .forEach(controller -> controller.actions()
-        .forEach(action -> {
-          // Rest and servlet filter WS should not be executed by the web service engine
-          if (shouldBeExecutedByWebServiceEngine(controller, action)) {
-            includeUrls.add("/" + controller.path() + "/*");
-          } else {
-            excludeUrls.add("/" + action.path() + "*");
-          }
-        }));
+    this.includeUrls = Stream.concat(
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/ws/WebServiceFilter.java]: 👍

File: server/sonar-server/src/main/java/org/sonar/server/user/ws/CreateAction.java [hunk 6]
```diff
@@ -60,6 +63,7 @@ public void define(WebService.NewController controller) {
       .setDescription("Create a user.<br/>" +
         "If a deactivated user account exists with the given login, it will be reactivated.<br/>" +
         "Requires Administer System permission<br/>" +
+        "Since 6.3, the password is only mandatory when creating local users, and should not be set on none local users<br/>" +
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/user/ws/CreateAction.java]: on non local users

File: server/sonar-server/src/main/java/org/sonar/server/ws/WebServiceFilter.java [hunk 7]
```diff
@@ -21,42 +21,49 @@
...
 
 import static org.sonar.server.property.ws.PropertiesWs.CONTROLLER_PROPERTIES;
 
 /**
  * This filter is used to execute Java WS.
  *
- * If the url match a Java WS, the output of the WS is returned and no other filers are executed.
- * If the url doesn't match a Java WS, then it's calling remaining filters (for instance to execute Rails WS).
+ * Every urls beginning with '/api' and every web service urls are taking into account,
+ * expect the ones that are directly implemented as a servlet filter and expect deprecated '/api/properties'.
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/ws/WebServiceFilter.java]: Ok, I will create a list, no there's no other domain.

**Comment by julienlancelot** [comment]
Merged on master

**Pull request closed without merging**